packages/components/pie-button/src/defs.ts

```diff
@@ -1,30 +1,36 @@
+export interface ButtonProps {
+    size: 'xsmall' | 'small-expressive' | 'small-productive' | 'medium' | 'large';
+    type: 'submit' | 'button' | 'reset' | 'menu';
+    variant: 'primary' | 'secondary' | 'outline' | 'ghost';
+}
+
 /**
  * Button size variants
  */
-export enum BUTTON_SIZE {
-    XSMALL = 'xsmall',
-    SMALL_EXPRESSIVE = 'small-expressive',
-    SMALL_PRODUCTIVE = 'small-productive',
-    MEDIUM = 'medium',
-    LARGE = 'large'
-}
+export const buttonSizes: ButtonProps['size'][] = [
+    'xsmall',
+    'small-expressive',
+    'small-productive',
+    'medium',
+    'large'
+];
```

Comment:
I think this could benefit from adding an `as const` assertion.
```ts
export const buttonSizes: ButtonProps['size'][] = [
    'xsmall',
    'small-expressive',
    'small-productive',
    'medium',
    'large'
] as const;
```
But I'm not completely sure.

Reply:
my code editor doesn't seem to like it 😬 
![image](https://github.com/justeattakeaway/pie/assets/24740015/dbbb0aef-4930-4d48-80d8-27d666e7688e)
I'll keep it as it is for now. But we can change it later if we need to?